tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -246,6 +278,48 @@ static int convertLLVMToSPIRV(const SPIRV::TranslatorOpts &Opts) {
           (SPIRV::SPIRVUseTextFormat ? kExt::SpirvText : kExt::SpirvBinary);
   }
 
+  if (SPIRVToolsDis) {
+#ifdef LLVM_SPIRV_HAVE_SPIRV_TOOLS
+    auto DisMessagePrinter = [](spv_message_level_t Level, const char *source,
+                                const spv_position_t &position,
+                                const char *message) -> void {
+      errs() << source << ": " << message << "\n";
+    };
+    spvtools::SpirvTools SpvTool(SPV_ENV_OPENCL_2_0);
+    SpvTool.SetMessageConsumer(DisMessagePrinter);
+    std::string DisOutput;
+
+    // Serialize the SPIR-V module to a uint32_t array and then invoke the
+    // SPIR-V tools disassembler to obtain textual assembly.
+    std::string Err;
+    StreambufToArray OutStreamBuf;
+    std::ostream OutStream(&OutStreamBuf);
+    bool Success = writeSpirv(M.get(), Opts, OutStream, Err);
+    if (!Success) {
+      errs() << "Failed to serialize SPIR-V: " << Err << '\n';
```

Comment:
Basically `writeSpirv` does 2 things: translates LLVM to SPIR-V and serializes it. AFAIK, we see errors from translation stage way more often then from serialization.

Reply:
Good point, I have rephrased it.